internal/jujuapi/applicationoffers_test.go

```diff
@@ -281,64 +282,61 @@ func (s *applicationOffersSuite) TestListApplicationOffers(c *gc.C) {
-		defer conn.Close()
-		client := applicationoffers.NewClient(conn)
-
-		results, err := client.Offer(
-			s.Model.UUID.String,
-			"test-app",
-			[]string{s.endpoint.Name},
-			"bob@canonical.com",
-			"test-offer1",
-			"test offer 1 description",
-		)
-		c.Assert(err, gc.Equals, nil)
-		c.Assert(results, gc.HasLen, 1)
-		c.Assert(results[0].Error, gc.IsNil)
-
-		offerURL := "bob@canonical.com/model-1.test-offer1"
-
-		err = client.RevokeOffer(auth.Everyone, "read", offerURL)
-		c.Assert(err, jc.ErrorIsNil)
-
-		err = client.GrantOffer("test.user@canonical.com", "unknown", offerURL)
-		c.Assert(err, gc.ErrorMatches, `"unknown" offer access not valid`)
-
-		err = client.GrantOffer("test.user@canonical.com", "read", "no-such-offer")
-		c.Assert(err, gc.ErrorMatches, `application offer not found`)
-
-		err = client.GrantOffer("test.user@canonical.com", "admin", offerURL)
-		c.Assert(err, jc.ErrorIsNil)
-
-		testUser := dbmodel.User{
-			Username: "test.user@canonical.com",
-		}
-
-		offer := dbmodel.ApplicationOffer{
-			URL: offerURL,
-		}
-		err = s.JIMM.Database.GetApplicationOffer(ctx, &offer)
-		c.Assert(err, jc.ErrorIsNil)
-		c.Assert(offer.UserAccess(&testUser), gc.Equals, "admin")
-
-		err = client.RevokeOffer("test.user@canonical.com", "consume", offerURL)
-		c.Assert(err, jc.ErrorIsNil)
-
-		err = s.JIMM.Database.GetApplicationOffer(ctx, &offer)
-		c.Assert(err, jc.ErrorIsNil)
-		c.Assert(offer.UserAccess(&testUser), gc.Equals, "read")
-
-		conn3 := s.open(c, nil, "user3")
-		defer conn3.Close()
-		client3 := applicationoffers.NewClient(conn3)
-
-		err = client3.RevokeOffer("test.user@canonical.com", "read", offerURL)
-		c.Assert(err, gc.ErrorMatches, "unauthorized")
-	*/
+	ctx := context.Background()
+
+	conn := s.open(c, nil, "bob@canonical.com")
+	defer conn.Close()
+	client := applicationoffers.NewClient(conn)
+
+	results, err := client.Offer(
+		s.Model.UUID.String,
+		"test-app",
+		[]string{s.endpoint.Name},
+		"bob@canonical.com",
+		"test-offer1",
+		"test offer 1 description",
+	)
+	c.Assert(err, gc.Equals, nil)
+	c.Assert(results, gc.HasLen, 1)
+	c.Assert(results[0].Error, gc.IsNil)
+
+	offerURL := "bob@canonical.com/model-1.test-offer1"
+
+	err = client.RevokeOffer(ofganames.EveryoneUser, "read", offerURL)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = client.GrantOffer("test.user@canonical.com", "unknown", offerURL)
+	c.Assert(err, gc.ErrorMatches, `"unknown" offer access not valid`)
+
+	err = client.GrantOffer("test.user@canonical.com", "admin", offerURL)
+	c.Assert(err, jc.ErrorIsNil)
+
+	testUser := openfga.NewUser(
+		&dbmodel.Identity{
+			Name: "test.user@canonical.com",
+		},
+		s.OFGAClient,
+	)
+
+	offer := dbmodel.ApplicationOffer{
+		URL: offerURL,
+	}
+	err = s.JIMM.Database.GetApplicationOffer(context.Background(), &offer)
+	c.Assert(err, gc.Equals, nil)
+
+	c.Assert(testUser.GetApplicationOfferAccess(ctx, offer.ResourceTag()), gc.Equals, ofganames.AdministratorRelation)
```

Comment:
I thought we were not going to in line func calls i.e., from our discussion on c.Assert(mythingreturnserr, gc.IsNil)? This seems like the same thing?

Reply:
fixed